Gems/PhysX/Code/Source/System/PhysXSystem.cpp

```diff
@@ -97,7 +99,20 @@ namespace PhysX
             m_systemConfig = *physXConfig;
         }
 
-        AzFramework::AssetCatalogEventBus::Handler::BusConnect();
+        // If the settings registry isn't available, something earlier in startup will report that failure.
+        if (auto* settingsRegistry = AZ::SettingsRegistry::Get();
```

Comment:
I'm wondering if we should roll this all up into a helper function? I'm seeing the if block + register handler + comments get copy and pasted all over.

Reply:
I see what you mean, it's one of those cases that it's a bit verbose to call `RegisterHandle`, but it's only one function and it's already a wrapper :) Another helper function around the wrapper seems too much. I'm open to suggestions though.